plugins/services/src/js/components/modals/NewCreateServiceModal.js

```diff
@@ -629,6 +597,48 @@ class NewCreateServiceModal extends Component {
     return [];
   }
 
+  getResetState(nextProps = this.props) {
+    const {location, params} = nextProps;
+    const isEdit = this.isLocationEdit(location);
+    const serviceID = decodeURIComponent(params.id || '/');
+    const service = isEdit ? DCOSStore.serviceTree.findItemById(serviceID) : null;
+    const isSpecificVersion = service instanceof Application && params.version;
+    let serviceConfig = new Application(
+      Object.assign({id: getBaseID(serviceID)}, DEFAULT_APP_SPEC)
+    );
+
+    if (isEdit && service instanceof Service && !isSpecificVersion) {
+      serviceConfig = service.getSpec();
+    }
+
+    if (isEdit && isSpecificVersion) {
+      serviceConfig = service.getVersions().get(params.version);
+    }
+
+    const newState = {
```

Comment:
_Side Note: We need to refactor the `NewCreateServiceModal` as there is a high chance of bugs with this number of interdependent state props._